src/feature-management/FeatureFlagsProvider.tsx

```diff
@@ -26,6 +26,54 @@ export const FeatureFlagsProvider = ({children} : Props): React.ReactNode => {
 
     setFlagState({...flagState, loading: true})
 
+    /**
+     * CUSTOM PROPERTIES
+     *
+     * Custom properties allow you to target specific users or segments in CloudBees platform.
+     * Set these BEFORE calling Rox.setup() so they're available for targeting rules.
+     *
+     * Types:
+     * - setCustomStringProperty: For text values (userId, email, plan, etc.)
+     * - setCustomNumberProperty: For numeric values (age, score, count, etc.)
+     * - setCustomBooleanProperty: For true/false values (isPremium, isAdmin, etc.)
+     * - setCustomDateProperty: For date/time values (registrationDate, lastLogin, etc.)
+     *
+     * Use in CloudBees platform:
+     * Create targeting rules like:
+     * - "Show feature if userTier = 'premium'"
+     * - "Show feature if registrationDate > '2024-01-01'"
+     */
+
+    // String property example - User tier for subscription-based targeting
+    Rox.setCustomStringProperty('userTier', () => {
+      return 'premium'; // variants: 'free', 'premium', 'enterprise'
+    });
+
+    // Number property example - User account age in days
+    Rox.setCustomNumberProperty('accountAgeInDays', () => {
+      return 45; // User registered 45 days ago
+    });
+
+    // Boolean property example - Beta tester status
+    Rox.setCustomBooleanProperty('isBetaTester', () => {
+      return true;
+    });
+
+    // Date property example - User registration date
+    Rox.setCustomDateProperty('registrationDate', () => {
+      // In production, get this from user's actual registration date
+      // Example: return new Date(user.registeredAt);
+      return new Date('2024-01-15'); // User registered on Jan 15, 2024
```

Comment:
Do you think we could easily introduce something like a `user` object? Then this could feel a bit more real because it would be set to `user.signUpDate` or something. (Obviously the focus is the SDK API, so don't go building a user management/login type system. Just hard-code it somewhere with a comment saying "this would have come from the login process..." or something).